File: src/main/java/org/ichnaea/respository/ProjectRepository.java
package org.ichnaea.respository;

import org.ichnaea.model.Project;
import org.ichnaea.model.User;

import java.util.List;

public interface ProjectRepository extends PersistentEntityRepository<Project> {

    List<User> findMembers(Long id);

    List<Project> findProjects(Long id);

    void addMember(Long id, Long userId);

    void removeMember(Long id, Long userId);
}

File: src/main/java/org/ichnaea/respository/ProjectRepository.kt
package org.ichnaea.respository

import org.ichnaea.model.Project
import org.ichnaea.model.User

object ProjectRepository : ListRepository<Project>() {
    init {

        val owner = User("admin", "")

        save(
            Project(
                name = "Ichnaea",
                description = "It is a Bug Tracker Project which has many bugs! " +
                        "Ironic. It could save others from bugs, but not itself.",
                owner = owner,
            )
        )

        save(
            Project(
                name = "Code for Good",
                description = "Showcase your tech skills and work alongside a team, guided by our technologists," +
                        " to solve real-world problems for social good organizations.",
                owner = owner,
            )
        )

    }
}